core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -116,9 +119,53 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
 
   override def deleteEntities(entityRefs: Seq[AttributeEntityReference],
                               parentContext: RawlsRequestContext
-  ): Future[Int] = ???
+  ): Future[Int] =
+    repository.dataSource.inTransaction { _ =>
+      for {
+        entities: Seq[Option[CompactEntityRecord]] <- DBIO.sequence(
+          entityRefs.map(ref => repository.queries.getEntity(workspaceId, ref.entityType, ref.entityName))
+        )
+        entityIds = entities.flatten.map(_.id).toSet
+        // check if any of these entities are referenced by someone else
+        referencingEntities: Seq[AttributeEntityReference] <- repository.queries.getReferencingEntities(entityIds)
+        _ = if (referencingEntities.toSet.diff(entityRefs.toSet).size != 0) {
+          throw new DeleteEntitiesConflictException(referencingEntities.toSet)
+        }
+        // remove all references from these entities
+        _ <- repository.queries.deleteAllReferences(entityIds)
+        res <- repository.queries.batchHide(workspaceId, entityRefs)
+      } yield res.sum
+    }
```

Comment:
instead of retrieving all the entities-to-be-deleted just to get their ids, then checking for references to those ids, could you build a single SQL statement? I'd love to see the database doing all the work instead of round-tripping back and forth between the db and Scala. It would be something like this:
```sql
select e.id, e.entity_type, e.name
from ENTITY e, ENTITY_REFS r
where r.from_id = e.id
and r.to_id in ( select id from ENTITY where ${types and names clause for entityRefs argument} )
```
You could potentially use a [WITH clause](https://dev.mysql.com/doc/refman/8.0/en/with.html) if you find that makes the SQL more readable.

Reply:
Using a with clause, it could look something like:
```sql
with
  -- select the id/type/name of all entities the user is asking to delete
  entitiesToBeDeleted as select ...
-- select the type/name of all entities which reference the entities-to-be-deleted,
-- and which are themselves not in entities-to-be-deleted
select e.entity_type, e.name
from ENTITY e, ENTITY_REFS r
where r.from_id = e.id
and r.to_id in (select id from entitiesToBeDeleted)
and r.from_id not in (select id from entitiesToBeDeleted);
```

Reply:
Ok i've [updated](https://github.com/broadinstitute/rawls/pull/3291/commits/65f337930fbc24a2305169b4a94782cfc51d4cd2) the queries and their use, but if I'm interpreting the execution plans correctly, the queries themselves seem a bit more costly, which makes sense since I'm doing joins in the query instead of multiple queries.  Is that going to be an improvement over multiple db calls?